### ISSUE TITLE
Only one service is created when creating a DB connected service

### ISSUE THREAD

**Issue by RalfEggert** [zfcampus/zf-apigility-admin-ui#57]
I have a database with two tables. I configured the database adapter and start to create a new API with two services, one for each table. When I choose DB connected service and select my database adapter, I see both tables. When I check both, I get an error `Error saving field`. Looking at the generated code the connection and entity classes are created for both services. But the configuration for the second service is missing. When I try to create the service for the second table afterwards, I get the errors `I cannot create the DB-Connected service, please check if already exists` and `Error creating DB-Connected service(s)`.

When I delete the API and recreate it again by creating a DB Connected service for one table and then another one for the second table afterwards, everything works fine. 

If this error depend on my database structure, here is a dump:

```
SET FOREIGN_KEY_CHECKS=0;
SET SQL_MODE = "NO_AUTO_VALUE_ON_ZERO";

DROP TABLE IF EXISTS `address`;
CREATE TABLE IF NOT EXISTS `address` (
  `id` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  `firstname` varchar(64) NOT NULL,
  `lastname` varchar(64) NOT NULL,
  `street` varchar(64) NOT NULL,
  `postcode` varchar(10) NOT NULL,
  `city` varchar(64) NOT NULL,
  `country` enum('de','at','ch','it') NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COMMENT='Address data' AUTO_INCREMENT=5 ;

DROP TABLE IF EXISTS `customer`;
CREATE TABLE IF NOT EXISTS `customer` (
  `id` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  `changed` datetime NOT NULL,
  `invoice_address` smallint(5) unsigned NOT NULL,
  `delivery_address` smallint(5) unsigned NOT NULL,
  PRIMARY KEY (`id`),
  KEY `address_id` (`invoice_address`),
  KEY `delivery_address` (`delivery_address`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 COMMENT='Customer data' AUTO_INCREMENT=5 ;

ALTER TABLE `customer`
  ADD CONSTRAINT `customer_ibfk_1` FOREIGN KEY (`invoice_address`) REFERENCES `address` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
  ADD CONSTRAINT `customer_ibfk_2` FOREIGN KEY (`delivery_address`) REFERENCES `address` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION;

SET FOREIGN_KEY_CHECKS=1;
```

**Comment by RalfEggert** [comment]
Another hint that might help: with Doctrine connected services this problem does not exist. Everything works fine

**Comment by pratheeshchambeth** [comment]
I have similar issue while creating DB Connected Service. I get an error 'Unable to fetch Service' .
Also the 'General Settings' tab for the service is blank.
However if I try to access the service using Postman, I do get proper response.

**Comment by ezimuel** [comment]
@RalfEggert I just tried with the master branch of *zf-apigility-admin-ui* and *zf-apigility-admin*, using your database structure and everything works fine. I'm able to create 2 services with DB connected feature and select each one from the tree view:

![issue_57](https://cloud.githubusercontent.com/assets/475967/8412286/314e9160-1e89-11e5-913f-ca1b5470e15b.png)

I suggest to try with the master branches of  *zf-apigility-admin-ui* and *zf-apigility-admin*. Anyway, we are working on Apigility 1.2 that will contains a lot of bug fixes, and we'll release soon.
/cc @pratheeshchambeth

**Comment by epoleacov** [comment]
Same error comes on *dev-master* and *dev-develop*, also tried with *zf-apigility-doctrine* without success, any ideas how to fix it?

**Comment by epoleacov** [comment]
Found hint: if i try to use table without underscore symbols "_" it works fine.

**Comment by colinkelly** [comment]
A fix was added for this issue #59. However, it got overwritten, there is another pull request that is pending for a merge #100
